packages/table/src/table.tsx

```diff
@@ -917,22 +930,47 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
     private columnHeaderCellRenderer = (columnIndex: number) => {
         const props = this.getColumnProps(columnIndex);
 
-        const { id, loadingOptions, renderCell, renderColumnHeader, ...spreadableProps } = props;
+        // TODO: `renderColumnHeader` is really an unclear name. should rename
+        // to `renderColumnHeaderCell` in the future.
+        // (see: https://github.com/palantir/blueprint/issues/1625)
+        const {
+            id,
+            loadingOptions,
+            renderCell,
+            renderColumnHeader: renderColumnHeaderCell,
+            ...spreadableProps,
+        } = props;
 
         const columnLoading = this.hasLoadingOption(loadingOptions, ColumnLoadingOption.HEADER);
 
-        if (renderColumnHeader != null) {
-            const columnHeader = renderColumnHeader(columnIndex);
-            const columnHeaderLoading = columnHeader.props.loading;
+        // <Table>'s useInteractionBar defaults to undefined. this means we
+        // won't override the cell's useInteractionBar value unless the consumer
+        // explicitly provided an override value to the <Table>.
+        const tableUseInteractionBar = this.props.useInteractionBar;
+
+        if (renderColumnHeaderCell != null) {
+            const columnHeaderCell = renderColumnHeaderCell(columnIndex);
+            const columnHeaderCellLoading = columnHeaderCell.props.loading;
+
+            // print a deprecation warning here if needed. we can't do this in
+            // ColumnHeaderCell, because we wouldn't be able to tell if
+            // useInteractionBar was injected by Table or provided by the user.
+            if (columnHeaderCell.props.useInteractionBar && !CoreUtils.isNodeEnv("production")) {
```

Comment:
@gscshoyru @themadcreator @llorca FYI I'm printing a deprecation warning from here in the `table.tsx` instead from of `ColumnHeaderCell.validateProps`. Do the comments above explain why clearly enough?

Reply:
Yes, that makes sense to me